### ISSUE TITLE
📚 Documentation: Why aren't the rate limits editable since it's self-hosted? What telemetric data is sent to AppWrite?

### ISSUE THREAD

**Issue by DankMemeGuy** [appwrite/appwrite#4965]
### 💭 Description

Since AppWrite is self-hosted, why are[ the rate limits ](https://appwrite.io/docs/rate-limits) imposed hardcoded? Why can't they be changed in the config? Every server is different, and every service is different. If I'm running a web server with 256 GB of RAM, very powerful CPU, NVME SSD, and a secured user system that would discourage abuse, I don't see why this is a hardcoded limit.

Furthermore, there is a statement on that page that says:
`To protect the quality of service from AppWrite, additional rate limits may apply to some actions.`

What does that mean? "Quality of service from AppWrite"? Is there data being sent to AppWrite? What data is this? I haven't seen any mention in the docs anywhere of telemetry data.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by mariomurrent-softwaresolutions** [comment]
Yeah it would be nice if the rate limit could be changes via the UI as well

**Comment by Meldiron** [comment]
Hey there 👋 Some rate limits exists to protect your 3rd party services, while others are there to protect your server directly.

For example, rate limit on Magic URL login is there to protect your SMTP server, as you are usually billed per each email that is sent.
Limit on `createDocument` is there purely to protect your database so 1 user cannot spam you with billions of documents overnight.
Another good example of rate limit is the one for registering user. As most rate limits are scoped to per-user when autorized, there has to be a rate limit for registering user too othersise other rate limits would not be relevant.

With all of that said, I agree with you that most of those could and should be configurable. I also agree with @mariomurrent-softwaresolutions suggesting this configuration should be in UI, as this should be done per-project.

Looking at other security settings, in my opinion, even variable like `_APP_OPTIONS_ABUSE` should not be env variable, instead, a toggle in the UI. That way you can enable/disable abuse protection for specific project in specific time, as nessessary.

**Comment by stnguyen90** [comment]
For configurable rate limits, please make sure to 👍🏼 this issue: https://github.com/appwrite/appwrite/issues/2953. I'm going to close this as a duplicate.
